projects/fsastorefrontlib/src/model/custom-product.interface.ts

```diff
@@ -0,0 +1,7 @@
+import { Product } from '@spartacus/core';
+
+export interface CustomProduct extends Product {
```

Comment:
`FSProduct` has everything except `salesIllustrationDiagramData`, so I would suggest just adding it to `FSProduct`. Also please try to avoid `any` when defining an interface.